### DISCUSSION TITLE
Subsystems and API Structures

### DISCUSSION THREAD

**Discussion by MTsimon** [zephyrproject-rtos/zephyr#62684] [Q&A]
I couldn't find an accurate answer during my search on Google and here on GitHub.

How can I determine which subsystems a driver has implemented without delving into the driver's source code and manually examining its API structure?

Some drivers handle multiple subsystems. For instance, a device might function as both an EEPROM and a sensor.

Is there any method to achieve this without having to inspect the driver's source code directly?

**Comment by MTsimon** [comment]
up

**Reply by yashi** [reply]
What do you expect? Doc? Run time? A west command?

Also would you mind why you need it or your use case please.

**Reply by MTsimon** [reply]
I was thinking about documentation. There is no other way, to lookup which subsystem a driver is using, other than digging into the source code of the specific driver.

Use case: I have NFC which also has an internal eeprom. The driver I wrote, is using the eeprom subsystem along with some proprietary functions, but I dont know where to document how to use the driver? should it be a readme inside the drivers directory, should I create an example of how to use the driver?

**Reply by yashi** [reply]
something like this?
https://docs.zephyrproject.org/latest/samples/drivers/clock_control_litex/README.html
